shell/platform/linux/public/flutter_linux/fl_plugin_registry.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_PLUGIN_REGISTRY_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_PLUGIN_REGISTRY_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+#include "fl_plugin_registrar.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_INTERFACE(FlPluginRegistry,
+                    fl_plugin_registry,
+                    FL,
+                    PLUGIN_REGISTRY,
+                    GObject)
+
+/**
+ * FlPluginRegistry:
+ *
+ * #FlPluginRegistry is FIXME
+ */
+
+struct _FlPluginRegistryInterface {
+  GTypeInterface g_iface;
+
+  /**
+   * FlPluginRegistry::get_registrar_for_plugin:
+   * @registry: an #FlPluginRegistry
+   * @name: plugin name
+   *
+   * Gets the plugin registrar for the the plugin with @name.
+   *
+   * Returns: (transfer full): an #FlPluginRegistrar
+   */
+  FlPluginRegistrar* (*get_registrar_for_plugin)(FlPluginRegistry* registry,
+                                                 const gchar* name);
+
+  /**
+   * FlPluginRegistry::add_plugin:
+   * @registry: an #FlPluginRegistry
+   * @plugin: an #FlPlugin
+   *
+   * Adds @plugin to the registry.
+   */
+  void (*add_plugin)(FlPluginRegistry* registry, FlPlugin* plugin);
```

Comment:
I don't think we'll need this. The C++ registrar (not registry; not sure why it's here?) has an 'add' because without reference counting something needs to own the plugins, and the registrar was a convenient place to assign ownership. In a ref-counted model, I would expect that channel handlers would ensure plugin lifetime (as happens on iOS and macOS)